src/test/java/org/apache/commons/compress/compressors/DetectCompressorTest.java

```diff
@@ -164,7 +183,70 @@ public void testDetect() throws Exception {
     }
 
     @Test
-    public void testDetection() throws Exception {
+    public void testDetectNullOrEmptyCompressorNames() throws Exception {
+        try (CompressorInputStream bzip2 = createStreamFor("bla.txt.bz2")) {
+            assertThrows(IllegalArgumentException.class, () -> {
+                CompressorStreamFactory.detect(bzip2, (Set<String>) null);
+            });
+        }
+
+        try (CompressorInputStream gzip = createStreamFor("bla.tgz")) {
+            assertThrows(IllegalArgumentException.class, () -> {
+                CompressorStreamFactory.detect(gzip, new HashSet<>());
+            });
+        }
+    }
+
+    @Test
+    public void testDetectLimitedByName() throws Exception {
+        assertEquals(CompressorStreamFactory.BZIP2, detect("bla.txt.bz2",
```

Comment:
Use longer lines (160), let's avoid "newspaper" formatting. I'll have to create a checktyle file for this component...

Reply:
All of it :-)